<a href="https://colab.research.google.com/github/Allajka/statistics/blob/main/seminar_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from math import factorial as fl

In [ ]:
def C (n, k):
  return fl(n) / (fl(k)*fl(n - k))

##**Задача 1**

Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150.

In [ ]:
values = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]
average = sum(values)/ len(values)

**1 Cреднее арифметическое: 1306 / 20 = 65.3**

*Несмещенная оценка дисперсии*
$$S = \sqrt{\frac{Σ(x_i -  μ)^2 * n_i} { n -1}}$$
*Cмещенная оценка дисперсии*
$$S = \sqrt{\frac{Σ(x_i -  μ)^2 * n_i} { n}}$$

In [ ]:
print(f"Сумма частот Σ = {(len(values))}")
print(f"Среднее μ = {average}")

Сумма частот Σ = 20
Среднее μ = 65.3


In [ ]:
values = [i**2 for i in values]

In [ ]:
average_mean = sum(values)/len(values)
dispersion_bias = average_mean - average **2

**2 Cмещенная дисперсия = 950.11**

In [ ]:
values = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
np.sum((values - average)**2) / (len(values) - 1)

1000.1157894736842

**3 Несмещенная дисперсия = 1000.11**

In [ ]:
standard_deviation = dispersion_bias**0.5

**4 Cреднее квадратичное отклонение = 30.823**

## Задача 2
В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

P = P(1) * P(2) + P(3) * Р(4) + P(5) * Р(6)

P(1) - 1 белый и 1 черный из первой

P(2) - 2 белых и 2 черный из второй

P(3) - 2 белых и 0 черный из первой

P(4) - 1 белый и 3 черный из второй

P(5) - 0 белых и 2 черных из первой

P(6) - 3 белых и 1 черный из второй

$P(1) =  \frac{C_{5}^{1} * C_{3}^{1} }{C_{8}^{2} }$ 
$P(2) =  \frac{C_{5}^{2} * C_{7}^{2} }{C_{12}^{4} }$
$P(3) =  \frac{C_{5}^{2} * C_{3}^{0} }{C_{8}^{2} }$
$P(4) =  \frac{C_{5}^{1} * C_{7}^{3} }{C_{12}^{4} }$
$P(5) =  \frac{C_{5}^{0} * C_{3}^{2} }{C_{8}^{2} }$
$P(6) =  \frac{C_{5}^{3} * C_{7}^{1} }{C_{12}^{4} }$

In [ ]:
P_1 =  (C(5, 1) * C(3, 1)) / C(8, 2)

In [ ]:
P_2 =  (C(5, 2) * C(7, 2)) / C(12, 4)

In [ ]:
P_3 =  (C(5, 2) * C(3, 0)) / C(8, 2)

In [ ]:
P_4 =  (C(5, 1) * C(7, 3)) / C(12, 4)

In [ ]:
P_5 =  (C(5, 0) * C(3, 2)) / C(8, 2)

In [ ]:
P_6 =  (C(5, 3) * C(7, 1)) / C(12, 4)

In [ ]:
P = P_1 * P_2 + P_3 * P_4 + P_5 * P_6
print(P)

0.3686868686868687


**Ответ: Вероятность того, что 3 мяча белые - 0.368**

## Задача 3
На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6.

**Формула Байеса**

$ P (B|A)= \frac{P (A|B) * P (B)} {P (A)} $

где, B - спортсмен A - попадание в мешень

In [ ]:
P_A = 1/3 * 0.9 + 1/3 * 0.8 + 1/3 * 0.6
P1 = (1/3 * 0.9) / P_A
P2 = (1/3 * 0.8) / P_A
P3 = (1/3 * 0.6) / P_A
print(P1, P2, P3)

0.391304347826087 0.3478260869565218 0.2608695652173913


**Ответ:**

a) вероятность того, что выстрел произведен первым спортсменом = 0.391

б) вероятность того, что выстрел произведен вторым спортсменом = 0.347

в) вероятность того, что выстрел произведен третьим  спортсменом = 0.260

## Задача 4
В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. 

**Формула Байеса**

$ P (B|A)= \frac{P (A|B) * P (B)} {P (A)} $

где, B - факультет A - сдача сессии

In [ ]:
PA = 1/4 * 0.8 + 1/4 * 0.7 + 1/2 * 0.9
P_1 = 1/4 * 0.8 / PA
P_2 = 1/4 * 0.7 / PA
P_3 = 1/2 * 0.9 / PA
print(P_1, P_2, P_3)

0.24242424242424246 0.21212121212121213 0.5454545454545455


**Ответ:** 

Студент сдал первую сессию. Какова вероятность, что он учится:

a) на факультете A = 0.242

б) на факультете B = 0.212

в) на факультете C = 0.545

## Задача 5 
Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. 

In [ ]:
p1, p2, p3 = 0.1, 0.2, 0.25

q1 = 1 - p1
q2 = 1 - p2
q3 = 1 - p3

**а) Вероятность того, что в первый месяц выйдут из строя все детали = 0.005**

$$P(X_3) = P(A_1) * P(A_2) * P(A_3) = p_1 * p_2 * p_3 $$

In [ ]:
PX3 = p1 * p2 * p3
print(PX3)

0.005000000000000001


**б) Вероятность того, что в первый месяц выйдут из строя только две детали = 0.08**  

Две детали сломаны, а одна - работает

$$P(X_2) = P(A_1) * P(A_2) * P(\overline{A}_3) + P(A_1) * P(\overline{A}_2) * P(A_3) + P(\overline{A}_1) * P(A_2) * P(A_3)= p_1 * p_2 * q_3 + p_1 * q_2 * p_3 + q_1 * p_2 * p_3 $$

In [ ]:
PX2 = p1 * p2 * q3 + p1 * q2 * p3 + q1 * p2 * p3
print(PX2)

0.08000000000000002


**в) Вероятность того, что в первый месяц выйдут из строя хотя бы одна деталь = 0.375**  

Две детали работают, а одна - сломана

$$P(X_1) = P(A_1) * P(\overline{A}_2) * P(\overline{A}_3) + P(\overline{A}_1) * P(\overline{A}_2) * P(A_3) + P(\overline{A}_1) * P(A_2) * P(\overline{A}_3)= p_1 * q_2 * q_3 + q_1 * q_2 * p_3 + q_1 * p_2 * q_3 $$

In [ ]:
PX1 = p1 * q2 * q3 + q1 * q2 * p3 + q1 * p2 * q3

print(PX1)

0.37500000000000006


**г) Вероятность того, что в первый месяц выйдут из строя от одной до двух деталей = 0.455**

Одна сломана, две работают **ИЛИ** две сломаны и одна работает

$$P(X_(1*2)) = P(X_1) + P(X_2) $$

In [ ]:
PX1_2 = PX1 + PX2
print(PX1_2)

0.45500000000000007
